### PR TITLE
[Impeller] Statically check stage interfaces for precision mismatches.

### PR DESCRIPTION
**Pull request by chinmaygarde**
Fixes https://github.com/flutter/flutter/issues/147078

### CODE REVIEW

**Comment by chinmaygarde** [comment]
This should trip on the bug @jonahwilliams just fixed. The matching doesn't work on the headers generated from Metal because the same semantics aren't applied.

**Comment by chinmaygarde** [comment]
Ok, I see the following failures. @jonahwilliams already fixed the fast gradient.

```
<impeller::PorterDuffBlendVertexShader, impeller::PorterDuffBlendFragmentShader>
<impeller::FastGradientVertexShader, impeller::FastGradientFragmentShader>
<impeller::TextureUvFillVertexShader, impeller::TiledTextureFillExternalFragmentShader>
``

**Comment by chinmaygarde** [comment]
I fixed up the shaders with precision values that made sense to me. PTAL.

**Review by jonahwilliams — APPROVED** [review]
LGTM!

File: impeller/renderer/shader_stage_compatibility_checker.h [hunk 1]
```diff
@@ -47,7 +47,8 @@ class ShaderStageCompatibilityChecker {
               input_slot->bit_width != output_slot->bit_width ||
               input_slot->vec_size != output_slot->vec_size ||
               input_slot->columns != output_slot->columns ||
-              input_slot->offset != output_slot->offset) {
+              input_slot->offset != output_slot->offset ||
+              input_slot->relaxed_precision != output_slot->relaxed_precision) {
```
Comment (gaaclarke) [inline impeller/renderer/shader_stage_compatibility_checker.h:51]: nice, thanks

**Pull request merged**